src/adapters/sharethrough.js

```diff
@@ -0,0 +1,132 @@
+var utils = require('../utils.js');
+var bidmanager = require('../bidmanager.js');
+var bidfactory = require('../bidfactory.js');
+
+const STR_BIDDER_CODE = "sharethrough";
+const STR_VERSION = "0.1.0"; //Need to check analytics too for version
+
+var SharethroughAdapter = function SharethroughAdapter() {
+
+  const str = {};
+  str.STR_BTLR_HOST = document.location.protocol + "//btlr.sharethrough.com";
+  str.STR_TEST_HOST =  document.location.protocol + "//btlr-prebid-test.sharethrough.com";
+  str.STR_BEACON_HOST = document.location.protocol + "//b.sharethrough.com/butler?";
+  str.placementCodeSet = {};
+
+  function _callBids(params) {
+    const bids = params.bids;
+
+    $$PREBID_GLOBAL$$.onEvent('bidWon', str.bidWon);
+
+    addEventListener("message", _receiveMessage, false);
+    
+    // cycle through bids
+    for (let i = 0; i < bids.length; i += 1) {
+      const bidRequest = bids[i];
+      str.placementCodeSet[bidRequest.placementCode] = bidRequest;
+      const scriptUrl = _buildSharethroughCall(bidRequest);
+      str.loadIFrame(scriptUrl);
+    }
+  }
+
+  function _buildSharethroughCall(bid) {
+    const testPkey = 'test';
+    const pkey = utils.getBidIdParameter('pkey', bid.params);
+
+    let host = (pkey === testPkey) ? str.STR_TEST_HOST : str.STR_BTLR_HOST;
+
+    let url = host + "/header-bid/v1?";
+    url = utils.tryAppendQueryString(url, 'bidId', bid.bidId);
+
+    if(pkey !== testPkey) {
+      url = utils.tryAppendQueryString(url, 'placement_key', pkey);
+      url = utils.tryAppendQueryString(url, 'ijson', '$$PREBID_GLOBAL$$.strcallback');
+      url = appendEnvFields(url);
+    } else {
+      url = url.substring(0, url.length - 1);
+    }
+
+    return url;
+  }
+
+  str.loadIFrame = function(url) {
+    const iframe = document.createElement("iframe");
+    iframe.src = url;
+    iframe.style.cssText = 'display:none;';
+
+    document.body.appendChild(iframe);
+  };
+
+  function _receiveMessage(event) {
+    if(event.origin === str.STR_BTLR_HOST || event.origin === str.STR_TEST_HOST) {
+      try {
+        $$PREBID_GLOBAL$$.strcallback(JSON.parse(event.data).response);
```

Comment:
this doesn't need to a public function because it's scoped within the adapter itself. Only use a public function if you use JSONP.